rust/lance/src/dataset/transaction.rs

```diff
@@ -744,7 +744,9 @@ impl Transaction {
             Operation::Restore { .. } => {
                 unreachable!()
             }
-            Operation::UpdateConfig { .. } => {}
+            Operation::UpdateConfig { .. } => {
+                final_fragments.extend(maybe_existing_fragments?.clone());
+            }
```

Comment:
Maybe we should combine this with the `Operation::ReserveFragments()` branch since they are both metadata-only updates. That way we can keep the logic up-to-date more easily.

Reply:
good call, made this change